### PR TITLE
Improve container routing / bump react vite / improve vite dev experience

### PR DESCRIPTION
**Pull request by frenautvh**
| Question         | Answer
| ---------------- | ---
| Branch?          | develop
| Bug fix?         | no
| New feature?     | yes
| Breaking change? | yes
| Tickets          | MANAGER-9806
| License          | BSD 3-Clause

## Description

- Rewrite routing for shell application with better capabilities (application path + application hash)
- Bump React in container to v18
- Bump vite to v3
- Add hot module reload capabilities in dev for both container & contained application

### CODE REVIEW

**Comment by antleblanc** [comment]
:eyes:  #7901

**Review by antleblanc — CHANGES_REQUESTED** [review]
@frenautvh Could you please resolve existing conflict?
Thank you!

File: packages/manager/core/vite-config/package.json [hunk 1]
```diff
@@ -0,0 +1,22 @@
+{
+  "name": "@ovh-ux/manager-vite-config",
+  "version": "0.0.0",
+  "private": "true",
```
Comment (marie-j) [inline packages/manager/core/vite-config/package.json:4]:
```suggestion
  "private": true,
```

File: packages/manager/core/vite-config/package.json [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+    "type": "git",
+    "url": "git+https://github.com/ovh/manager.git",
+    "directory": "packages/manager/core/vite-config"
+  },
+  "license": "BSD-3-Clause",
+  "author": "OVH SAS",
+  "type": "module",
+  "main": "./src/index.js",
+  "dependencies": {
+    "@ovh-ux/manager-dev-server-config": "^3.0.0 || ^4.0.0",
```
Comment (marie-j) [inline packages/manager/core/vite-config/package.json:16]:
```suggestion
    "@ovh-ux/manager-dev-server-config": "^4.0.0",
```

**Review by marie-j — CHANGES_REQUESTED** [review]

**Review by Jisay — DISMISSED** [review]

**Review by cbourgois — DISMISSED** [review]

**Review by Jisay — APPROVED** [review]

File: packages/manager/modules/config/package.json [hunk 3]
```diff
@@ -33,8 +33,5 @@
   "dependencies": {
     "@ovh-ux/ovh-reket": "^1.0.0",
     "lodash-es": "^4.17.15"
-  },
-  "devDependencies": {
-    "typescript": "^4.3.2"
   }
-}
\ No newline at end of file
+}
```
Comment (marie-j) [inline packages/manager/modules/config/package.json:37]: Isn't typescript not needed anymore ? I still see `.ts` files in this module

**Review by Jisay — APPROVED** [review]

**Pull request merged**